src/components/nav/nav.tsx

```diff
@@ -0,0 +1,116 @@
+ * @slot progress - A slot for adding a `calcite-progress` component to the primary nav level.
+ * @slot primary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the primary nav level.
+ * @slot primary-content-center - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the center position of the primary nav level.
+ * @slot primary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the primary nav level.
+ * @slot secondary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the secondary nav level.
+ * @slot secondary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the secondary nav level.
+ * @slot tertiary-content-start - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the start position of the tertiary nav level.
+ * @slot tertiary-content-end - A slot for adding a `calcite-menu`, `calcite-action`, or other interactive elements in the end position of the tertiary nav level.
+ */
+export class CalciteNav {
+  //--------------------------------------------------------------------------
+  //
+  //  Element
+  //
+  //--------------------------------------------------------------------------
+  @Element() el: HTMLCalciteNavElement;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Properties
+  //
+  //--------------------------------------------------------------------------
+
+  /**
+   * When true, display a hamburger  icon and emits `calciteNavMenuActionSelect` event on user interaction.
+   */
+  @Prop({ reflect: true }) displayMenuAction = false;
+
+  /**
+   * Specifies the label of the hamburger icon.
+   */
+  @Prop({ reflect: true }) label!: string;
+
+  // --------------------------------------------------------------------------
+  //
+  //  Events
+  //
+  // --------------------------------------------------------------------------
+
+  /**
+   * Emits whenever the component is selected or unselected.
+   *
+   */
+  @Event() calciteNavMenuActionSelect: EventEmitter<void>;
+
+  //--------------------------------------------------------------------------
+  //
+  //  Private Methods
+  //
+  //--------------------------------------------------------------------------
+
+  private clickHandler = () => {
+    this.calciteNavMenuActionSelect.emit();
+  };
+
+  //--------------------------------------------------------------------------
+  //
+  //  Render Methods
+  //
+  //--------------------------------------------------------------------------
+
+  renderMenuAction(): VNode {
+    return <calcite-action icon="hamburger" onClick={this.clickHandler} text={this.label} />;
```

Comment:
I would prefer allowing a user to slot this so they can modify the action properties like indicator, active, etc.

Reply:
Discussed in other comment - seems like providing this + a slot of the same name that overrides the property seems to work.